api/server/router/container/container_routes.go

```diff
@@ -446,10 +428,6 @@ func (s *containerRouter) wsContainersAttach(ctx context.Context, w http.Respons
 	}
 	containerName := vars["name"]
 
-	if !s.backend.Exists(containerName) {
-		return derr.ErrorCodeNoSuchContainer.WithArgs(containerName)
```

Comment:
Understood. I was looking at this before and got stuck in the same place. The problem is the call to the daemon/backend is a callback, and not a call in the current context. The only way I can see is to move the websocket stuff inside of the daemon/backend, but that is again moving some http context inside of the daemon.